crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -341,10 +338,11 @@ impl<'a, Provider: ResolverProvider> Resolver<'a, Provider> {
                 ResolverVersion::Available(version) => version,
                 ResolverVersion::Unavailable(version, unavailable) => {
                     let reason = match unavailable {
-                        UnavailableVersion::RequiresPython(requires_python) => {
-                            // Incompatible requires-python versions are special in that we track
-                            // them as incompatible dependencies instead of marking the package version
-                            // as unavailable directly
+                        // Incompatible requires-python versions are special in that we track
+                        // them as incompatible dependencies instead of marking the package version
+                        // as unavailable directly
+                        UnavailableVersion::IncompatibleDist(IncompatibleDist::Source(IncompatibleSource::RequiresPython(requires_python))
+| IncompatibleDist::Wheel(IncompatibleWheel::RequiresPython(requires_python))) => {
```

Comment:
This ensures we do not regress as described in https://github.com/astral-sh/uv/pull/2066#discussion_r1506843879

Reply:
The formatting here seems off.